integrated_channels/integrated_channel/transmitters/learner_data.py

```diff
@@ -32,6 +32,86 @@ def __init__(self, enterprise_configuration, client=IntegratedChannelApiClient):
             client=client
         )
 
+    def single_learner_assessment_grade_transmit(self, exporter, **kwargs):
+        """
+        Placeholder
+        """
+        TransmissionAudit = apps.get_model(  # pylint: disable=invalid-name
+            app_label=kwargs.get('app_label', 'integrated_channel'),
+            model_name=kwargs.get('model_name', 'LearnerDataTransmissionAudit'),
+        )
+        kwargs.update(
+            TransmissionAudit=TransmissionAudit,
+        )
+
+        # Even though we're transmitting a single learner, they can have records per assessment (multiple per course)
+        # and there can potentially be one learner within multiple configurations
+        for learner_data in exporter.single_assessment_level_export(**kwargs):
+            serialized_payload = learner_data.serialize(enterprise_configuration=self.enterprise_configuration)
+            try:
+                code, body = self.client.create_course_completion(
+                    getattr(learner_data, kwargs.get('remote_user_id')),
+                    serialized_payload
+                )
+            except ClientError as client_error:
+                code = client_error.status_code
+                body = client_error.message
+                self.handle_transmission_error(learner_data, client_error)
+
+            learner_data.status = str(code)
+            learner_data.error_message = body if code >= 400 else ''
+
+            learner_data.save()
+
+    def assessment_level_transmit(self, exporter, **kwargs):
+        """
+        Placeholder
+        """
+        TransmissionAudit = apps.get_model(  # pylint: disable=invalid-name
+            app_label=kwargs.get('app_label', 'integrated_channel'),
+            model_name=kwargs.get('model_name', 'LearnerDataTransmissionAudit'),
+        )
+        kwargs.update(
+            TransmissionAudit=TransmissionAudit,
+        )
+
+        # Retrieve learner data for each existing enterprise enrollment under the enterprise customer
+        # and transmit the data according to the current enterprise configuration.
+        for learner_data in exporter.bulk_assessment_level_export(**kwargs):
```

Comment:
this is my stupidity, but in the bulk_assessment_level_export method we're yielding records, which from my understanding suspends the function. At what point does it produce all results for this for loop? Is it basically streaming them (ie 'if record, yield record' goes to 'for learner_data in that record...' then go back to the suspended bulk function where we left off, yield another record, etc.?)

Reply:
Don't think it's stupidity at all, I think your description is fairly accurate. It's just a way to stream each learner data record to the transmitter.